### PR TITLE
add blog on localstack x shipyard

### PR DESCRIPTION
**Pull request by HarshCasper**

### CODE REVIEW

File: content/2024-04-01-ephemeral-environments-with-localstack-shipyard/index.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+-   Improve the development process for cloud infrastructure by allowing developers to test code changes in isolation and iterate quickly.
+-   Save time and money by eliminating the need to wait for staging to be configured and blocking other team members during the process.
+-   Facilitate collaboration across different teams by providing a freshly-configured pre-production environment for everyone to work in.
+-   Run your automated end-to-end (E2E) test suite and pre-seed your infrastructure & application state to allow more thorough testing.
+-   Streamline the login process (SSO) for your team to prevent anyone outside your organization from accessing your environments.
+    
+
+## Conclusion
+
+Congratulations! You’ve successfully deployed an AWS-powered cloud application on an ephemeral environment using LocalStack & Shipyard. With LocalStack, you don’t have to worry about deleting your ephemeral environment after a quick round of testing — LocalStack’s core emulation layer allows you to persist the resources created on the host machine for extensive testing and further use Cloud Pods to pre-seed the infrastructure state as and when required!
```
Comment (nbeck415) [inline content/2024-04-01-ephemeral-environments-with-localstack-shipyard/index.md:251]: maybe we clarify this a little further "worry about deleting your ephemeral environment" -- are we worried about losing resources after deletion? maybe we say "losing [x] after spinning down your ephemeral environments"

**Review by nbeck415 — COMMENTED** [review]
This is really good! Just made a few small suggestions. Lmk if you need help getting images for this!

File: content/2024-04-01-ephemeral-environments-with-localstack-shipyard/index.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+Click on the **Add environment variables** to go to the next step, and add any environment variables that you want to configure. Finally, click on **Create application** to get started with the first build of your application!
+
+<graphic-showing-the-build-queue.png>
+
+After a successful build, you can now click on the **Visit** button on your application dashboard to navigate to the deployed application. You can start interacting with the application, and review the run logs for your application services.
+
+<graphic-showing-the-localstack-run-logs.png>
+
+### Create a GitHub Action workflow for previews
```
[the remark this answers is not in the document]
Comment (HarshCasper) [reply content/2024-04-01-ephemeral-environments-with-localstack-shipyard/index.md:187]: Can we change this to `### Create PR comments for your ephemeral environments`?

**Review by nbeck415 — APPROVED** [review]
LGTM!

File: content/2024-04-01-ephemeral-environments-with-localstack-shipyard/index.md [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+-   Improve the development process for cloud infrastructure by allowing developers to test code changes in isolation and iterate quickly.
+-   Save time and money by eliminating the need to wait for staging to be configured and blocking other team members during the process.
+-   Facilitate collaboration across different teams by providing a freshly-configured pre-production environment for everyone to work in.
+-   Run your automated end-to-end (E2E) test suite and pre-seed your infrastructure & application state to allow more thorough testing.
+-   Streamline the login process (SSO) for your team to prevent anyone outside your organization from accessing your environments.
+    
+
+## Conclusion
+
+Congratulations! You’ve successfully deployed an AWS-powered cloud application on an ephemeral environment using LocalStack & Shipyard. With LocalStack, you don't have to worry about losing your infrastructure state after spinning down your ephemeral environments after a quick round of testing. LocalStack allows you to persist the resources created on the host machine for extensive testing and further use Cloud Pods to pre-seed the infrastructure state as and when required!
```
Comment (whummer) [inline content/2024-04-01-ephemeral-environments-with-localstack-shipyard/index.md:286]:
```suggestion
Congratulations! You’ve successfully deployed an AWS-powered cloud application on an ephemeral environment using LocalStack & Shipyard. With LocalStack, you don't have to worry about losing your infrastructure state after tearing down your ephemeral environments after a quick round of testing. LocalStack allows you to persist the resources created on the host machine for extensive testing and further use Cloud Pods to pre-seed the infrastructure state as and when required!
```

File: content/2024-04-01-ephemeral-environments-with-localstack-shipyard/index.md [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+
+You can further explore Shipyard and their offering for various use cases such as:
+
+-   Adding Datadog logging and integrating with LocalStack’s serverless resources, such as Lambda, EventBridge, SQS, and more.
+-   Using Shipyard’s volume management to revert the data in case of regression and load them across sibling environments.
+-   Monitoring key infrastructure metrics such as an overview of build and deploy times, CPU/Memory usage, and deployment timeline.
+
+If you have any questions about configuring and running your project, drop by Shipyard Slack Community or LocalStack Slack Community. We would love to hear your feedback about this integration!
```
Comment (whummer) [inline content/2024-04-01-ephemeral-environments-with-localstack-shipyard/index.md:294]:
```suggestion
If you have any questions about configuring and running your project, drop by Shipyard's or LocalStack's Slack Community. We would love to hear your feedback about this integration. Happy cloud testing!
```

**Review by whummer — APPROVED** [review]
Looks awesome, kudos @HarshCasper and @nbeck415 for putting together this sample app that nicely showcases the integration between Shipyard & LocalStack! 🚀 

Added a few minor comments/suggestions, but nothing major. Great write-up - looking forward to seeing this published!

**Comment by nbeck415** [comment]
> Looks awesome, kudos @HarshCasper and @nbeck415 for putting together this sample app that nicely showcases the integration between Shipyard & LocalStack! 🚀
> 
> Added a few minor comments/suggestions, but nothing major. Great write-up - looking forward to seeing this published!

Thanks for the review @whummer! And thanks @HarshCasper for such a great writeup!

**Pull request merged**